visualisations/fan_chart/webviz_fan_chart/__init__.py

```diff
@@ -223,14 +228,15 @@ def __init__(self, data, observations=None):
                 else:
                     raise ValueError('An unknown column was passed: ', column)
 
-        for i, row in self.observations.iterrows():
-            lines.append(add_observation(row))
-            if row['name'] in uniquelines:
-                lines.append(
-                    add_marker(
-                        row,
-                        format_color(colors[row['name']], 1)
+        if validate_observation_data(self.observations):
```

Comment:
This seems rather strange, Likely you would want to throw a `ValueError` if the data is not valid.

Reply:
I do, inside the validator function